src/utils/viewer/viewer.cpp

```diff
@@ -878,16 +937,41 @@ void Viewer::drawEvent() {
     // reset timeSinceLastSimulation, accounting for potential overflow
     timeSinceLastSimulation = fmod(timeSinceLastSimulation, 1.0 / 60.0);
   }
-  uint32_t visibles = 0;
-  if (flyingCameraMode_) {
+
```

Comment:
It'd be nice to see these ~25 lines encapsulated to a helper function, just to keep viewer.cpp a bit more readable.

Reply:
Hello, welcome to our discussion about "to inline or not inline". :)

In `FisheyeShader.h`:
https://github.com/facebookresearch/habitat-sim/pull/1131/files#r589273848

Reply:
@mosra 's concerns in that other thread were around unnecessary use of virtual. Here, I'm just asking for a helper method in the Viewer class. I don't think this is a controversial ask.

Reply:
When I gave the link, I simply provided a pointer to the beginning of the lengthy conversation. You need to read through it. "virtual" is simply one part but not the main topic in it. Most of the discussion was focusing on "inline or not inline".

Reply:
Ok, I skimmed the Carmack article link. The situation here is basically a state machine (fisheye, flying, else) where I'm arguing that each state's logic should be encapsulated. That's not the same as Carmack's code (below), so I don't think Carmack would particularly advocate for inlining all these different states' logic.
```
void MajorFunction( void ) {
        MinorFunction1();
        MinorFunction2();
        MinorFunction3();
}
```

Anyway, I don't feel strongly. This is my two cents. I won't block the PR on this point.

Reply:
>I'm arguing that each state's logic should be encapsulated.

I agree.

@eundersander :  I was actually on your side. :) 
I remembered I did the similar thing when I developed the object picking functions in the viewer and @mosra requested me to inline everything. See [here](https://github.com/facebookresearch/habitat-sim/pull/661#discussion_r463000522). 

But I guess @mosra would require me to rollback if I do it. :grin:

Reply:
This is not my code, I don't *require* anybody to do anything ;) 

Just stating my opinion and sharing past experience.